jsk_perception/include/jsk_perception/histogram_of_oriented_gradients.h

```diff
@@ -1,44 +1,44 @@
+// Copyright (C) 2016 by Krishneel Chaudhary @ JSK Lab, The University
+// of Tokyo
 
 #ifndef _HISTOGRAM_OF_ORIENTED_GRADIENTS_H_
 #define _HISTOGRAM_OF_ORIENTED_GRADIENTS_H_
 
-// OpenCV Header Directives
 #include <opencv2/opencv.hpp>
 #include <opencv2/highgui/highgui.hpp>
 #include <opencv2/imgproc/imgproc.hpp>
-
-#include <string>
-#include <vector>
+#include <opencv2/features2d/features2d.hpp>
 
 class HOGFeatureDescriptor {
 
-    //  HOG Configuration Params
-#define N_BINS 9
-#define ANGLE 180.0
-#define BINS_ANGLE (ANGLE / N_BINS)
-#define CELL 8
-#define BLOCK 2
-
+ protected:
+    int N_BINS;
+    int ANGLE;
+    int BINS_ANGLE;
+    int CELL;
+    int BLOCK;
+    int num_threads_;
+    
  private:
-    virtual void bilinearBinVoting(
+    virtual void histogramBinVoting(
        const float &, int &, int &);
     virtual void imageGradient(
        const cv::Mat &, cv::Mat &);
     virtual cv::Mat blockGradient(
-       const int, const int, cv::Mat &);
-    virtual cv::Mat orientationistogram(
-       const cv::Mat&, const int &, const int &, bool = false);
+        const int, const int, const int, cv::Mat &);
     virtual void getHOG(
        const cv::Mat &, cv::Mat &, cv::Mat &);
-    template<typename T>
-    T computeHOGHistogramDistances(
-       const cv::Mat &, std::vector<cv::Mat> &,
-       const int = CV_COMP_BHATTACHARYYA);
    
  public:
-    HOGFeatureDescriptor();
-    virtual cv::Mat computeHOG(
-       const cv::Mat &);
+    HOGFeatureDescriptor(
+        const int = 8, const int = 2, const int = 9,
+        const float = 180.0f, const int = 1);
+    virtual cv::Mat computeHOG(const cv::Mat &);
+    template<typename T>
+    T computeHOGHistogramDistances(
+        const cv::Mat &, std::vector<cv::Mat> &,
+        const int = CV_COMP_BHATTACHARYYA);
+    cv::Mat visualizeHOG(
+        cv::Mat &, cv::Mat &, cv::Size, int, double);
```

Comment:
Not right now

--
◉ Kei Okada

2016/04/18 23:21、Krishneel <notifications@github.com> のメッセージ:

> In jsk_perception/include/jsk_perception/histogram_of_oriented_gradients.h:
> 
> >     
> >   public:
> > -    HOGFeatureDescriptor();
> > -    virtual cv::Mat computeHOG(
> > -       const cv::Mat &);
> > +    HOGFeatureDescriptor(
> > +        const int = 8, const int = 2, const int = 9,
> > +        const float = 180.0f, const int = 1);
> > +    virtual cv::Mat computeHOG(const cv::Mat &);
> > +    template<typename T>
> > +    T computeHOGHistogramDistances(
> > +        const cv::Mat &, std::vector<cv::Mat> &,
> > +        const int = CV_COMP_BHATTACHARYYA);
> > +    cv::Mat visualizeHOG(
> > +        cv::Mat &, cv::Mat &, cv::Size, int, double);
> I see. There were few changes to almost all the functions. Besides the spacing, should I change something else.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>